ToolBox/Benzinga/BenzingaNewsDataConverter.cs

```diff
@@ -0,0 +1,451 @@
+                    // write all of the indexes in one go per day
+                    indexCollection.Indexes.Add($"{article.Id}.json");
+
+                    createdReference = true;
+                }
+
+                // Skip if we didn't create any references for an article
+                if (!createdReference)
+                {
+                    Log.Error($"BenzingaDataConverter.SerializeArticlesAndIndexes(): Skipping news article {article.Id} because no tickers are associated with it");
+                    continue;
+                }
+
+                // Remove unwanted symbols from the article before we write to disk
+                foreach (var symbolToRemove in symbolsToRemove)
+                {
+                    article.Symbols = article.Symbols.Where(s => s != symbolToRemove).ToList();
+                }
+
+                // We could potentially have an article with no Symbols and still write it without this check
+                if (article.Symbols.Count == 0)
+                {
+                    Log.Error($"BenzingaDataConverter.FilterArticlesAndIndexes(): Skipping news article {article.Id} because there are no Symbols associated with this article");
+                    continue;
+                }
+
+                // Batch all the articles so that we can write it all in one shot
+                filtered.ArticleContents[$"{article.Id}.json"] = JsonConvert.SerializeObject(article, Formatting.None, new BenzingaNewsJsonConverter());
+            }
+
+            return filteredCollection;
+        }
+
+        /// <summary>
+        /// Writes all the filtered indexes from memory to disk
+        /// </summary>
+        private void WriteIndexesToFile(BenzingaNewsFiltered filtered)
+        {
+            foreach (var kvp in filtered.SymbolIndexes)
+            {
+                var symbol = kvp.Key;
+                var indexCollection = kvp.Value;
+
+                var referenceFileDirectory = new DirectoryInfo(Path.Combine(_destinationDirectory.FullName, symbol.Value.ToLowerInvariant()));
+                var referenceFile = new FileInfo(Path.Combine(referenceFileDirectory.FullName, $"{indexCollection.Date.ToStringInvariant(DateFormat.EightCharacter)}.csv"));
+
+                // Make sure to create the directory, otherwise we won't be able to move files to it.
+                referenceFileDirectory.Create();
+
+                var referenceFileExisted = referenceFile.Exists;
+                var referenceFileTemp = Path.Combine(Path.GetTempPath(), $"{Guid.NewGuid()}.csv");
+                var existingReferenceFileBackupPath = Path.Combine(Path.GetTempPath(), $"{Guid.NewGuid()}.csv");
+
+                Log.Trace($"BenzingaDataConverter.WriteToFile(): Creating reference temp file: {referenceFileTemp}");
+                // Order before writing since HashSet doesn't guarantee order of elements
+                File.WriteAllLines(referenceFileTemp, indexCollection.Indexes.OrderBy(x => x));
```

Comment:
Since we are using `UpdatedAt` as end time think `OrderBy(id)` isn't valid anymore? for example:
- article id 1: update at 16:05:10
- article id 2: update at 3:10:03